From: Bruce Rogers <brogers@suse.com>
Date: Tue, 16 Feb 2021 11:29:41 -0700
Subject: [ath5k] Add missing AR5K_EEPROM_READ in ath5k_eeprom_read_turbo_modes

Git-commit: 19d0fab40f07eeea7fe6b9e0d4e8d4b0c2de215f
References: boo#1181922

The GCC11 compiler pointed out something that apparently no previous
compiler noticed: in ath5k_eeprom_pread_turbo_modes, local variable
val is used uninitialized. From what I can see, the code is just
missing an initial AR5K_EEPROM_READ. Add it right before the switch
statement.

Signed-off-by: Bruce Rogers <brogers@suse.com>
Signed-off-by: Michael Brown <mcb30@ipxe.org>
---
 src/drivers/net/ath/ath5k/ath5k_eeprom.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/roms/ipxe/src/drivers/net/ath/ath5k/ath5k_eeprom.c b/roms/ipxe/src/drivers/net/ath/ath5k/ath5k_eeprom.c
index 983d206b7e36cc4981ecbb760a41..12519bc599fb7d4757da43a6bcc2 100644
--- a/roms/ipxe/src/drivers/net/ath/ath5k/ath5k_eeprom.c
+++ b/roms/ipxe/src/drivers/net/ath/ath5k/ath5k_eeprom.c
@@ -416,6 +416,7 @@ ath5k_eeprom_read_turbo_modes(struct ath5k_hw *ah,
 	if (ee->ee_version < AR5K_EEPROM_VERSION_5_0)
 		return 0;
 
+	AR5K_EEPROM_READ(o++, val);
 	switch (mode){
 	case AR5K_EEPROM_MODE_11A:
 		ee->ee_switch_settling_turbo[mode] = (val >> 6) & 0x7f;
